gcd/src/main/java/org/spine3/server/storage/datastore/DsRecordStorage.java

```diff
@@ -385,8 +386,12 @@ protected void writeRecords(Map<I, EntityRecord> records) {
     @Override
     public Iterator<I> index() {
         checkNotClosed();
-        final StructuredQuery.Filter filter = PropertyFilter.eq(TYPE_URL_PROPERTY_NAME, typeUrl.getTypeName());
-        return Indexes.indexIterator(datastore, KIND, idClass, filter);
+        return Indexes.indexIterator(datastore, kind, idClass);
+    }
+
+    @VisibleForTesting
+    String getKind() {
```

Comment:
Let's make a simple value object for `Kind`. It should be immutable and `final`, with a factory method, which validates for `null` etc.